addons/road-generator/nodes/road_intersection.gd

```diff
@@ -133,32 +138,38 @@ func add_branch(road_point: RoadPoint) -> void:
 	_sort_edges_clockwise()
 	emit_transform()
 
+
 ## Remove an edge from the intersection, sorting edges and updating the mesh afterwards.
 ## Does nothing if the edge is not present.
 func remove_branch(road_point: RoadPoint) -> void:
 	edge_points.erase(road_point)
 	_sort_edges_clockwise()
 	emit_transform()
 
+
 ## Sort edges, then refresh the mesh.
 func sort_branches() -> void:
 	_sort_edges_clockwise()
 	emit_transform()
 
+
 # ------------------------------------------------------------------------------
 #endregion
 #region Utilities
 # ------------------------------------------------------------------------------
 
+
 func refresh_intersection_mesh() -> void:
 	print("debug - refreshing intersection mesh")
 	if not is_instance_valid(settings) or not is_instance_valid(container):
 		return
 	if not container.create_geo:
 		return
 	
-	var mesh: Mesh = settings.generate_mesh(self.transform, edge_points)
+	var mesh: Mesh = settings.generate_mesh(self.transform, edge_points, container)
```

Comment:
FYI @Picorims on this adjustment. Open to other opinions, but in essence - we need to be able to access a number of settings on the container such as the materials and so forth. Even this `_create_collisions` function also applies many important features like the render and collision layers. 

To me, it seemed simplest to just pass in the container as another layer, but ultimately it still feels weird to not pass in the intersection (I know why we don't, circular imports etc). So, open to other ideas if you've got em.